activities/TamTamMicro.activity/app.js

```diff
@@ -47,33 +48,53 @@ enyo.kind({
 		enyo.forEach(this.$.collections.getControls(), function(item) { items.push(item); });
 		for (var i = 0 ; i < items.length ; i++) { items[i].destroy();	}
 
-		// Display collections
-		document.getElementById("body").style.backgroundColor = this.userColor.fill;
-		this.$.collections.applyStyle("background-color", this.userColor.fill);
-		var len = TamTam.collections.length;
-		for(var i = 0 ; i < len ; i++ ) {
-			this.$.collections.createComponent(
-				{ kind: "TamTam.Collection", name: TamTam.collections[i].name, selection: (i == this.collection), ontap: "changeCollection" },
-				{ owner: this }
-			).render();
-		}
-
 		// Remove items
 		var items = [];
 		this.$.items.applyStyle("background-color", this.userColor.fill);
 		enyo.forEach(this.$.items.getControls(), function(item) { items.push(item); });
 		for (var i = 0 ; i < items.length ; i++) { items[i].destroy();	}
+		if(this.pianoMode) {
+			document.getElementById("body").style.backgroundColor = this.userColor.fill;
+			
+			this.$.collections.createComponent(
+				{ kind: "TamTam.Collection", name: "piano", selection: (i == this.collection), ontap: "changePianoMode" },
+				{ owner: this }
+			).render();
+			
+			this.$.items.createComponent(
+				{ kind: "TamTam.Piano"},
+				{ owner: this }
+			).render();	
 
-		// Display items
-		var collection = TamTam.collections[this.collection];
-		var len = collection.content.length;
-		for(var i = 0 ; i < len ; i++ ) {
-			var item = this.$.items.createComponent(
-				{ kind: "TamTam.Item", name: collection.content[i] },
+		} else {
+			// Display collections
+			document.getElementById("body").style.backgroundColor = this.userColor.fill;
+			this.$.collections.applyStyle("background-color", this.userColor.fill);
+			var len = TamTam.collections.length;
+			for(var i = 0 ; i < len ; i++ ) {
+				this.$.collections.createComponent(
+					{ kind: "TamTam.Collection", name: TamTam.collections[i].name, selection: (i == this.collection), ontap: "changeCollection" },
+					{ owner: this }
+				).render();
+			}
+			
+			// TODO: get icon
+			this.$.collections.createComponent(
+				{ kind: "TamTam.Collection", name: "piano", selection: (i == this.collection), ontap: "changePianoMode" },
```

Comment:
This line generate a message in the browser console:

`
    enyo.js:21 enyo.Component.addComponent():  Duplicate component name "piano" in owner "app" violates unique-name-under-owner rule, replacing existing component in the hash and continuing, but this is an error condition and should be fixed.`

Seems related to the fact that you can't use piano as `name` because another component (probably the piano item) has already the same id.

Reply:
fb7d165